metalware-new/src/metal_log.rb

```diff
@@ -50,17 +48,31 @@ def metal_log
     end
 
     def initialize(log_name)
-      file = "#{self.class.config.log_path}/#{log_name}.log"
+      file = "#{config.log_path}/#{log_name}.log"
       FileUtils.mkdir_p File.dirname(file)
       f = File.open(file, "a")
       f.sync = true
       super(f)
-      level = self.class.config.log_serverity
+      self.level = config.log_severity
     end
 
     def warn(msg)
-      Output.stderr "warning: #{msg}"
-      super msg
+      if config.cli.strict
+        e = StrictWarningError.new(msg)
+        e.set_backtrace(caller(2))
```

Comment:
I think messing with backtraces in this way is a risky idea - it obfuscates the actual source of the error so is likely to make things harder to debug in future as we won't know the actual backtrace. We just display the error message to a user anyway rather than the full backtrace so I can't see an advantage to hiding where it came from to ourselves.

I'll remove this line then to try to avoid this possibility and then merge this through. Thanks for explaining :slightly_smiling_face:.